db/migrate/20210119154820_add_request_method_and_response_headers_to_vendor_api_requests.rb

```diff
@@ -0,0 +1,16 @@
+class AddRequestMethodAndResponseHeadersToVendorAPIRequests < ActiveRecord::Migration[6.0]
+  def change
+    add_column :vendor_api_requests, :request_method, :string
+    add_column :vendor_api_requests, :response_headers, :jsonb
+
+    VendorAPIRequest.where(request_method: nil).find_each do |vendor_api_request|
+      if vendor_api_request.request_path =~ /\/(offer|conditions-not-met|confirm-conditions-met|reject|confirm-enrolment|regenerate|generate|clear)$/
```

Comment:
A client could have made a `GET` to one of these endpoints in error. I'd almost rather have no data than unreliable data?

Reply:
@duncanjbrown in this case I think we'd serve a 404, we could omit any vendor api requests with 404 status from update?

Reply:
:+1: Good idea!